### DISCUSSION TITLE
Error in expm1(x = x) : non-numeric argument to mathematical function

### DISCUSSION THREAD

**Discussion by royfrancis** [satijalab/seurat#7232] [Q&A]
I randomly run into this error when running `DotPlot()`. It's hard to reproduce consistently. It kind of happens when number of identities are above 30. Has anyone experienced this before?

**Comment by sejjer7** [comment]
Yes, I have as well. Trying to plot 80 feature across 8 identities (which is probably too many but trying to illustrate something not have it totally readable). 

Update: I took out the code I had in to change x axis text size and angle and then it worked ok. But when I put it back in again I get the message again. So I've just made the plot as a gg object and then amended the text elements in a separate line of code

**Reply by royfrancis** [reply]
Sounds interesting! Would you be able to share the part of the code that is buggy?

**Reply by sejjer7** [reply]
If I do
DotPlot(Tcells, features = c("gene1", "gene2", "gene3".........)) + theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1.1, vjust = 1.1))
it brings up that error

But if I do
plot <-DotPlot(Tcell features = c("gene1", "gene2", "gene3".........))
plot + theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1.1, vjust = 1.1))
everything is fine

Obviously where I have put "genes" and "gene3".... this was actual gene names (89 of them)

I suppose that was messy code anyway, so to make things better I have made an character list from the Tcellmarkers list I had and got the gene names from

to_plot <- unique(Tcellmarkers)
DotPlot(Tcell, features = to_plot) + theme(axis.text.x = element_text(size = 6, angle = 45, hjust = 1.1, vjust = 1.1))

And all fine now.
